### ISSUE TITLE
Cloud Provider Discovery

### ISSUE THREAD

**Issue by cohix** [suborbital/grav#19]
Grav should be able to discover other nodes in a cloud provider using the provider's APIs.

**Comment by cohix** [comment]
https://github.com/hashicorp/go-discover

**Comment by denopink** [comment]
@cohix can I take this on?

**Comment by cohix** [comment]
@denopink yep! I'd suggest that we do a walkthrough together since Grav's Transport and Discovery plug-ins are not well documented.
